Lib/string.py

```diff
@@ -64,14 +71,24 @@ class Template:
     # See https://bugs.python.org/issue31672
     idpattern = r'(?a:[_a-z][_a-z0-9]*)'
     braceidpattern = None
-    flags = _re.IGNORECASE
+    flags = _sentinel_flags  # default: re.IGNORECASE
```

Comment:
Why not using `None` here?

Reply:
`flags` is meant to be overridden in subclasses, currently if someone uses `flags = None` it will raise an exception in `re.compile`, but if we check for `None` then the default flags will silently be used. 

That said, it has always failed, so perhaps it's just simpler to check for `None`?

Reply:
> currently if someone uses flags = None it will raise an exception in re.compile

I think it's not meant to be used as such. `flags` should be an integer and if it's not it's more "garbage in -> garbage out". If they want to disable the flag, they should set it to 0. Are the docs saying anything about the expected type of `flags`? I think it's fine to keep a None sentinel.